pkg/resource/cache_subnet_group/custom_update.go

```diff
@@ -0,0 +1,127 @@
+package cache_subnet_group
+
+import (
+	"context"
+
+	svcapitypes "github.com/aws-controllers-k8s/elasticache-controller/apis/v1alpha1"
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	svcsdk "github.com/aws/aws-sdk-go/service/elasticache"
+)
+
+func (rm *resourceManager) customUpdateCacheSubnetGroup(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (updated *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkUpdate")
+	defer func() {
+		exit(err)
+	}()
+	input, err := rm.newUpdateRequestPayload(ctx, desired, delta)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp *svcsdk.ModifyCacheSubnetGroupOutput
+	_ = resp
+	resp, err = rm.sdkapi.ModifyCacheSubnetGroupWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATE", "ModifyCacheSubnetGroup", err)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Message() {
+			case "No modifications were requested.":
+				return desired, nil
+			}
+		}
+		return nil, err
+	}
```

Comment:
IMO this is happening because the controller is trying to make an update request when it shouldn't. Idealy we should either fix the delta/readone first. And maybe eventually leverage [`OmitUnchangedField`](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/config/resource.go#L322-L326) to generate the update function

Reply:
Based on your suggestion, I investigated a bit more and found that the post set output template I added ([here](https://github.com/aws-controllers-k8s/elasticache-controller/pull/111)) was fixing the primary issue where the controller fails with a **Reconciler error** (`spec.subnetIDs: Required value`) before even starting the `Sync` and thus not hitting `sdkFind`. 
After the addition of a read many post set output template, now I see that the `Sync` starts with success and the template logic appends the `ko.Spec.SubnetIDs` slice with duplicate values which misleads the delta to which leads to updating the resource.

_My quick solution:_ I am just checking if `ko.Spec.SubnetIDs` is populated or not before proceeding.
_Another approach (perhaps):_ If there is another way to start `Sync` with success without a template addition, please advise, it could perhaps be a more sound approach 🤷

Reply:
I think we should not have duplicates at all - we need to focus on properly appending the subnets. Check my comment in the other PR https://github.com/aws-controllers-k8s/elasticache-controller/pull/111/files#r1133856530
@mustafasencer great job on testing and investigating this. I see that there were no elasticate releases 2 days ago but you still test the controllers - I'm curious to know how you do it :)

Reply:
@A-Hilaly Thank for the prompt response! I addressed the comment in the new commit.

I am testing locally via the kind cluster where I build only the local image of the controller and test the adopted resource scenario.